### PR TITLE
Fix SQLAlchemy 1.4 deprecation warnings

### LINKED ISSUE #46: SQLAlchemy 1.4 changes the signature of ConnectionEvents before_execute & after_execute
After upgrading to SQLAlchemy 1.4.22 (from 1.3.x), when using `ProfilingSession`, I now get warnings like:

> SADeprecationWarning: The argument signature for the "ConnectionEvents.before_execute" event listener has changed as of version 1.4, and conversion for the old argument signature will be removed in a future release.  The new signature is "def before_execute(conn, clauseelement, multiparams, params, execution_options)"

And:

> SADeprecationWarning: The argument signature for the "ConnectionEvents.after_execute" event listener has changed as of version 1.4, and conversion for the old argument signature will be removed in a future release.  The new signature is "def after_execute(conn, clauseelement, multiparams, params, execution_options, result)"

### PR DESCRIPTION
**Pull request by rsyring**
And other project tweaks to bring it more up-to-date.

This raises the minimum supported Python version to 3.6 and SQLAlchemy to 1.4.  Should those be documented somewhere?

fixes #46 

Todo:

- [ ] Make sure travis runs all tox envs.  If not, fix `[travis]` section in tox.ini.

### CODE REVIEW

**Comment by inconshreveable** [comment]
@rsyring thank you so much for this! can you summarize how this was tested? it looks like travis isn't working at the moment and giving us a green checkmark, any ideas what's going on there?

**Comment by rsyring** [comment]
@inconshreveable I tested locally by just running `tox`.

As to Travis, I really don't know anything about that.  I did see and update the Travis config file and assumed it would build automatically when the PR was created.  I don't use Travis so I'm not sure how that gets wired in.  Is there a dashboard in Travis for this project that can be checked?

I'm more familiar with CircleCI and can migration the CI to that if you want: https://circleci.com/docs/oss

**Comment by inconshreveable** [comment]
> @inconshreveable I tested locally by just running `tox`.
> 
> As to Travis, I really don't know anything about that. I did see and update the Travis config file and assumed it would build automatically when the PR was created. I don't use Travis so I'm not sure how that gets wired in. Is there a dashboard in Travis for this project that can be checked?
> 
> I'm more familiar with CircleCI and can migrate the CI workflow to that if you want: https://circleci.com/docs/oss

happy to migrate to circle if that's easier for you! let me know how i can help

**Comment by guruofgentoo** [comment]
@inconshreveable FYI, I work with @rsyring and have taken responsibility on our side for moving the branch for this PR forward.

The Travis CI problem most likely has to do with their changes for OSS projects last year. In order to put in the CircleCI config and have a working build for PR purposes, I have created a separate fork, grabbed Randy's branch, and put up a new PR intended to supersede this one (#49).

If that PR is merged, you'll need to add the project in CircleCI to automate builds for future updates. If you have any questions on that process, I'll be happy to discuss.
